src/main/java/jdbchomework/dao/jdbc/AbstractJdbcDao.java

```diff
@@ -0,0 +1,118 @@
+package jdbchomework.dao.jdbc;
+
+import jdbchomework.dao.model.GenericDao;
+import jdbchomework.entity.AbstractEntity;
+import org.slf4j.LoggerFactory;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+
+public abstract class AbstractJdbcDao<T extends AbstractEntity> implements GenericDao<T> {
+
+    private static org.slf4j.Logger log = LoggerFactory.getLogger(AbstractJdbcDao.class);
+    private String table;
+    private String column;
+
+    protected Connection connection;
+
+    public AbstractJdbcDao(Connection connection, String table, String column) {
+        this.connection = connection;
+        this.table = table;
+        this.column = column;
+    }
+
+    @Override
+    public boolean deleteById(long id) {
+        boolean result = false;
+        try (PreparedStatement statement = connection
+                .prepareStatement("DELETE FROM " + table + " WHERE " + column + " = ?;")) {
+            statement.setLong(1, id);
+            if (statement.executeUpdate() > 0) {
+                result = true;
+            }
+        } catch (SQLException e) {
+            log.error("Cannot connect to DB", e);
+            throw new RuntimeException(e);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS00112)